### PR TITLE
Physics refactor.

### PR DESCRIPTION
**Pull request by Richardn2002**
Rationale:
- `PhysicsState` and `PhysicsSystem` trait can be viewed as the requirements for the provided Runge-Kutta method impl to work. So they are all contained in file `runge_kutta.rs`. Simple Physics systems can choose to utilize the impl, thus using the structs and implementing the traits, or not, completely forgetting about that file, during implementation of `tick()`.
- `tick()` of `SimplePhysicsSystems` enum-dispatches to `tick()`s of variants, instead of dynamic dispatches, as for inox2d users are not supposed to bring their own physics systems (as of now).
- Decoupling is done realizing that one step of physics simulation is a closure yielding a `Vec2` and changing some contained states, per step. Thus letting one step do the calc, return the `Vec2` and letting external code setting params based on the return value will solve the previous ownership problem.
- TODO: The call to step the physics system is put naively inside `end_set_params()`, and without actual time elapsed passed in. Takeaway is, the physics step should be after param setting and before full puppet transform re-calc. But future refactor should provide the user with a better interface (physics still run if `end_set_params()` not called, and actual time elapsed passed in).

### CODE REVIEW

File: inox2d/src/physics.rs [hunk 1]
```diff
@@ -5,26 +5,78 @@ mod simple_physics;
...
+                ref mut bob,
+                ref mut state,
+            } => {
+                // Run the spring pendulum simulation
+                runge_kutta::tick(&spring_pendulum::eval, state, props, anchor, dt);
+
+                *bob = state.getv_bob();
+
+                *bob
+            }
```
Comment (Richardn2002) [inline inox2d/src/physics.rs]: Probably should put these into `pendulum.rs`.

File: inox2d/src/physics/pendulum.rs
Comment (Richardn2002) [inline inox2d/src/physics/pendulum.rs:1]: Two files for two types of pendulums please

File: inox2d/src/physics.rs (hunk 1 above)
Comment (Speykious) [reply inox2d/src/physics.rs]: I don't think so. This is where any physics system will tick, it's not pendulum-specific.

Comment (Speykious) [reply inox2d/src/physics.rs]: Or maybe you mean each individual tick strategy... If that's the case I'll need to come up with a struct to contain the state for each variant of `SimplePhysicsSystem`

Comment (Richardn2002) [reply inox2d/src/physics.rs]: I mean apparently two branches are not logically identical (one has the trigonometry while one does not) so they are pendulum-specific...?

Comment (Speykious) [reply inox2d/src/physics.rs]: So you *are* talking about the individual branches, ok got it

**Comment by Speykious** [comment]
I'll merge this in and we can continue the work on the branch itself

**Pull request merged**